modules/dnn/src/webgpu/include/tensor.hpp

```diff
@@ -0,0 +1,47 @@
+#ifndef OPENCV_DNN_WEBGPU_TENSOR_HPP
+#define OPENCV_DNN_WEBGPU_TENSOR_HPP
+// #ifdef HAVE_WEBGPU
+#include<dawn/webgpu_cpp.h>
+// #endif
+#include"buffer.hpp"
+#include"wgpucom.hpp"
+#include<memory>
+namespace cv { namespace dnn { namespace webgpu {
+// #ifdef HAVE_WEBGPU
+class Buffer;
+class Tensor{
```

Comment:
@wzw-intel , could you please help review this interface and its implementation?